helixbass/tsc-rust
# Parse + bind string literal

In this PR:
- parse + bind a string literal type + string literal value

To test:
If you run `cargo run tmp.tsx` against a source file which contains eg `const a: 'foo' = 'baz'`, you should see `post-binding:` debug-logging where the `VariableDeclaration`'s `type_` is a `LiteralTypeNode` whose `literal` is a `StringLiteral` with `text: "foo"` and the `VariableDeclaration`'s `initializer` is a `StringLiteral` with `text: "baz"` (and both should have `parent` links initialized)

Based on `type-check-generic-interface`
